sdk/core/Azure.Core/src/Diagnostics/AzureCoreEventSource.cs

```diff
@@ -317,5 +318,20 @@ private static string FormatHeaders(IEnumerable<HttpHeader> headers, HttpMessage
             }
             return stringBuilder.ToString();
         }
+
+        [NonEvent]
+        public void FailedToDecodeCaeChallengeClaims(string? encodedClaims, Exception exception)
+        {
+            if (IsEnabled(EventLevel.Error, EventKeywords.None))
+            {
+                FailedToDecodeCaeChallengeClaims(encodedClaims, exception.ToString());
+            }
+        }
+
+        [Event(FailedToDecodeCaeChallengeClaimsEvent, Level = EventLevel.Error, Message = "BearerTokenAuthenticationPolicy Failed to decode CAE claims: '{0}'. Exception: {1}")]
```

Comment:
@m-redding, any considerations we should have with regard to adding this to the AzureCoreEventSource that you're aware of?

Reply:
Nope! This looks good from my perspective